app/models/reference/CurrencyCode.scala

```diff
@@ -33,7 +35,18 @@ case class CurrencyCode(currency: String, description: String) extends Selectabl
 }
 
 object CurrencyCode {
-  implicit val format: OFormat[CurrencyCode] = Json.format[CurrencyCode]
+
+  def reads(config: FrontendAppConfig): Reads[CurrencyCode] =
+    if (config.isPhase6Enabled) {
+      (
+        (__ \ "key").read[String] and
+          (__ \ "value").read[String]
+      )(CurrencyCode.apply)
+    } else {
+      Json.reads[CurrencyCode]
+    }
+
+  implicit val format: Format[CurrencyCode] = Json.format[CurrencyCode]
```

Comment:
@jackheslop96 do we need to use the format method here, since we already take care of the reads manually, I believe all we need is the writes.

Reply:
The format can be used implicitly for both reads and writes to mongo. The `reads` def is only used explicitly in the ReferenceDataConnector